src/lib/permutations/ascon_perm/ascon_perm.h

```diff
@@ -0,0 +1,65 @@
+/*
+* Permutation Ascon_p[rounds] as specified in NIST SP.800-232, Section 3
+* (C) 2025 Jack Lloyd
+*     2025 René Meusel
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_ASCON_PERM_H_
+#define BOTAN_ASCON_PERM_H_
+
+#include <array>
+#include <cstdint>
+#include <optional>
+#include <span>
+#include <string>
+
+namespace Botan {
+
+using AsconState = std::array<uint64_t, 5>;
+
+struct AsconConfig {
+      uint8_t init_and_final_rounds;
+      uint8_t processing_rounds;
+      uint8_t bit_rate;
+      AsconState initial_state;
+};
+
+/**
+ * Ascon_p as specified in NIST SP.800-232, Section 3
+ */
+class Ascon_p final {
+   public:
+      constexpr explicit Ascon_p(AsconConfig config) :
+            m_S(config.initial_state),
+            m_bit_rate(config.bit_rate),
+            m_init_final_rounds(config.init_and_final_rounds),
+            m_processing_rounds(config.processing_rounds) {}
```

Comment:
Should have some basic checks of the input like rounds > 0 and <= 16, bitrate is possible, ...